### ISSUE TITLE
Dependency on Jackson even tho it is only used in one extension method

### ISSUE THREAD

**Issue by dtropanets** [rybalkinsd/kohttp#115]
Hey. I've noticed that there is a dependency on Jackson. Is it really that necessary to add it to the library just to use it inside of one extension method? Can it be possibly moved to a separate artifact?

**Comment by rybalkinsd** [comment]
@dtropanets that's a good point. I was also thinking about it some times ago.
Are you satisfied with the default configuration of `defaultHttpClient`?

Just in case, did you get any Jackson version conflicts while using?

Potentially, it's possible to move default _client configuration feature_ to the separate artifact.

**Comment by rybalkinsd** [comment]
As for the current vision:
- we should keep `ResponseBody.asJson()` ext function, however, provided by a specific artifact:
    - kohttp-fastjson    
    - kohttp-jackson
    - kohttp-moshi
- yaml configuration for default client should be implemented as a separate module
    - kohttp-config

**Comment by rybalkinsd** [comment]
@dtropanets do you have any further suggestions according to the vision

**Comment by rybalkinsd** [comment]
@gokulchandra You previously get in touch with json within the project. Would you be interested in PoC design for this feature?

**Comment by dtropanets** [comment]
@rybalkinsd We had to use custom client in order to use proxy, so can't tell much about the default one. Regarding the Jackson, I excluded all Jackson modules from dependency after we've noticed that it is included in 0.9.0. Don't remember having any issues. Also, can't tell much about yaml config, since I won't be able to use it, since I have to dynamically build the client in order to apply the proxy values received from the build machine (we use kohttp exclusively for CI scripts).
